catalog/plugin/catalog-plugin-clavin/src/main/java/org/codice/ddf/catalog/content/plugin/clavin/ClavinPlugin.java

```diff
@@ -0,0 +1,125 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.plugin.clavin;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import javax.activation.MimeType;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.bericotech.clavin.gazetteer.GeoName;
+import com.google.common.net.MediaType;
+
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.content.operation.CreateStorageResponse;
+import ddf.catalog.content.plugin.PostCreateStoragePlugin;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+
+public class ClavinPlugin implements PostCreateStoragePlugin {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClavinPlugin.class);
+
+    private ClavinWrapper clavinWrapper;
+
+    public void setClavinWrapper(ClavinWrapper clavinWrapper) {
+        this.clavinWrapper = clavinWrapper;
+    }
+
+    @Override
+    public CreateStorageResponse process(CreateStorageResponse input)
+            throws PluginExecutionException {
+
+        List<ContentItem> contentItems = input.getCreatedContentItems();
+        for (ContentItem contentItem : contentItems) {
+            if (applyTo(contentItem)) {
+                enrich(contentItem);
+            }
+        }
+        return input;
+    }
+
+    private void enrich(ContentItem contentItem) throws PluginExecutionException {
+        Metacard metacard = contentItem.getMetacard();
+        File file;
+        try {
+            file = contentItem.getFile();
+        } catch (IOException e) {
+            throw new PluginExecutionException("Failed to get file from metacard.", e);
+        }
+
+        // heuristic to filter out "meaningless" locations base on single char.
+        try {
+            clavinWrapper.parse(file)
+                    .stream()
+                    .filter(resolvedLocation -> resolvedLocation.getMatchedName()
+                            .length() > 1)
+                    .forEach(resolvedLocation -> {
+                        GeoName geoName = resolvedLocation.getGeoname();
+
+                        // lat,lon
+                        String lon = doubleToString(geoName.getLongitude());
+                        String lat = doubleToString(geoName.getLatitude());
+                        if (!StringUtils.isBlank(lon) && !StringUtils.isBlank(lat)) {
+                            String wkt = String.format("POINT(%s %s)", lon, lat);
+                            metacard.setAttribute(new AttributeImpl(Metacard.GEOGRAPHY, wkt));
```

Comment:
Shouldn't we be building a multipoint wkt instead of clobbering this attribute for each of the resolved locations?
`MULTIPOINT ((10 40), (40 30), (20 20), (30 10))`